scripts/dts-localize.ts

```diff
@@ -0,0 +1,261 @@
+    } else {
+      const filename = relative.split('/').join('-');
+      newPath = path.join(outputPath, filename);
+    }
+    return newPath;
+  }
+
+  function wasReplaceAlready(fileName: string, target: string) {
+    if (replaceRemapping.has(fileName) && replaceRemapping.get(fileName).includes(target)) {
+      console.log('skipped', target);
+      return true;
+    }
+    return false;
+  }
+
+  function getReplacementPath(
+    currentSourceFile: string,
+    referencedSourceFile: string,
+    target: string
+  ) {
+    const targetRelativeToSource2 = path
+      .relative(filesRemapping.get(currentSourceFile), filesRemapping.get(referencedSourceFile))
+      .slice(1)
+      .replace('.d.ts', '')
+      .replace('.ts', '');
+
+    replaceRemapping.set(currentSourceFile, [
+      ...(replaceRemapping.get(currentSourceFile) || []),
+      targetRelativeToSource2,
+    ]);
+
+    console.log('renamed', target);
+    return targetRelativeToSource2;
+  }
+
+  function wasIgnored(target: string) {
+    if (externals.includes(target)) {
+      console.log('ignored', target);
+
+      return true;
+    }
+    return false;
+  }
+
+  function replaceImport(node: ts.Node) {
+    if (
+      (ts.isImportDeclaration(node) || ts.isExportDeclaration(node)) &&
+      node.moduleSpecifier !== undefined
+    ) {
+      // @ts-ignore
+      const target: string = node.moduleSpecifier.text;
+      let currentSourceFile = '';
+      let referencedSourceFile = '';
+
+      if (wasIgnored(target)) {
+        return true;
+      }
+
+      currentSourceFile = node.getSourceFile().fileName;
+
+      if (wasReplaceAlready(currentSourceFile, target)) {
+        return true;
+      }
+
+      // find the sourceFile the import is pointing to
+      referencedSourceFile = getSourceFile(
+        typeChecker.getSymbolAtLocation(node.moduleSpecifier).valueDeclaration
+      ).fileName;
+
+      const replacementPath = getReplacementPath(currentSourceFile, referencedSourceFile, target);
+      // @ts-ignore
+      node.moduleSpecifier = ts.createStringLiteral(replacementPath);
+
+      return true;
+    }
+
+    if (ts.isImportTypeNode(node)) {
+      const target = node.argument.getText().slice(1, -1);
+      let currentSourceFile = '';
+      let referencedSourceFile = '';
+
+      if (wasIgnored(target)) {
+        return true;
+      }
+
+      currentSourceFile = node.getSourceFile().fileName;
+
+      if (wasReplaceAlready(currentSourceFile, target)) {
+        return true;
+      }
+
+      // find the sourceFile the import is pointing to
+      referencedSourceFile = getSourceFile(
+        typeChecker.getSymbolAtLocation(node).valueDeclaration
+      ).fileName;
+
+      const replacementPath = getReplacementPath(currentSourceFile, referencedSourceFile, target);
+
+      // @ts-ignore
+      node.argument = ts.createStringLiteral(replacementPath);
+      // node.argument = ts.factory.createStringLiteral(replacementPath); // TS4
+
+      return true;
+    }
```

Comment:
This seems very duplicative with the block above

Reply:
It is, but the way the target is read, and how to replacementPath is changed are different. How would you suggest I make this code let duplicative without adding more misdirection?

It'd require at least 3 extra if-else blocks, which might be just as hard to understand?